cloud/fallback-public-cloud.yaml

```diff
@@ -69,7 +69,7 @@ clouds:
   azure:
     type: azure
     description: Microsoft Azure
-    auth-types: [ interactive, service-principal-secret ]
```

Comment:
I'm a bit unsure about whether it's really necessary to have a separate auth-type for this. Could we not just have "interactive" use the Azure CLI if it's available, and fall back to doing the device code login otherwise? Maybe too big of a change for 2.2 at this stage.

Can we please drop the "-import", and just call it "azure-cli"? I don't think the `-import` really adds much for the user. Also, we might want to look into just running `az ad sp create-for-rbac`, which isn't exactly an "import".